internal/skillpuller/source/url.go

```
@@ -0,0 +1,120 @@
+
+	return &Skill{Name: name, Content: content}, nil
+}
+
+// rejectPrivateHost blocks requests to loopback, link-local, and
+// private IP ranges to prevent SSRF.
+func rejectPrivateHost(host string) error {
+	ip := net.ParseIP(host)
+	if ip == nil {
+		ips, err := net.LookupIP(host)
+		if err != nil || len(ips) == 0 {
+			return nil // let the HTTP client handle DNS failures
```

golangci-lint (2.12.2)

[error] error is not nil (line 75) but it returns nil

(nilerr)